nbgitpuller/pull.py

```diff
@@ -98,18 +104,21 @@ def repo_is_dirty(self):
         except subprocess.CalledProcessError:
             return True
 
+
     def update_remotes(self):
         """
         Do a git fetch so our remotes are up to date
         """
-        yield from execute_cmd(['git', 'fetch'], cwd=self.repo_dir)
+        yield from execute_cmd(['git', 'fetch', '--tags'], cwd=self.repo_dir)
+
 
     def find_upstream_changed(self, kind):
         """
         Return list of files that have been changed upstream belonging to a particular kind of change
         """
+        log_arg = 'HEAD..{}'.format(self.tag_name if self.tag_name else 'origin/{}'.format(self.branch_name))
```

Comment:
During `git log`, tags needs nothing ahead of them while branches needs `origin/` ahead of them.